google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -443,11 +444,21 @@ public boolean deleteLog(String log) {
     return get(deleteLogAsync(log));
   }
 
+  public boolean deleteLog(String log, LogDestinationName destination) {
+    return get(deleteLogAsyncImpl(log, destination));
+  }
+
   public ApiFuture<Boolean> deleteLogAsync(String log) {
-    DeleteLogRequest request =
-        DeleteLogRequest.newBuilder()
-            .setLogName(LogName.ofProjectLogName(getOptions().getProjectId(), log).toString())
-            .build();
+    return deleteLogAsyncImpl(log, null);
+  }
+
+  public ApiFuture<Boolean> deleteLogAsync(String log, LogDestinationName destination) {
+    return deleteLogAsyncImpl(log, destination);
```

Comment:
nit: why extra call to `deleteLogAsyncImpl` is needed? why not to keep implementation in `deleteLogAsync`?